app/src/main/java/swati4star/createpdf/activity/MainActivity.java

```diff
@@ -166,14 +167,32 @@ public boolean onCreateOptionsMenu(Menu menu) {
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         if (item.getItemId() == R.id.menu_favourites_item) {
+            Fragment currFragment = getSupportFragmentManager().findFragmentById(R.id.content);
+
             Fragment fragment = new FavouritesFragment();
             FragmentManager fragmentManager = getSupportFragmentManager();
             setTitle(R.string.favourites);
-            fragmentManager.beginTransaction().replace(R.id.content, fragment).commit();
+            FragmentTransaction transaction = fragmentManager.beginTransaction()
+                    .replace(R.id.content, fragment);
+            if (!(currFragment instanceof HomeFragment)) {
+                transaction.addToBackStack(getFragmentName(currFragment));
+            }
+            transaction.commit();
         }
         return super.onOptionsItemSelected(item);
     }
 
+    private String getFragmentName(Fragment fragment) {
+        String name = "set name";
+        if (fragment instanceof ViewFilesFragment) {
+            name = getString(R.string.viewFiles);
+        } else if (fragment instanceof HistoryFragment) {
+            name = getString(R.string.history);
+        }
+        // TODO : set name for other cases also
```

Comment:
@sidhuparas @vyankatesh24 @Swati4star 
back button is working good, but I have problem setting title. I can think of  doing it this way. can you suggest me a better way?

Reply:
@Ni3verma Use Hashmap. Check out `titleMap()` method of `MainActivity`.

Reply:
Here I need to check if the passed in fragment is an instance of some class.
How will I do this with Hashmap?
I think if, elseIf will be good way

Reply:
For now, it seems the way to go. However there need a new issue to be created because there are lots of places where instanceof trees have been used. I think this can serve the purpose: https://stackoverflow.com/a/18232037

Reply:
so should I use if, else-if tree now?

Reply:
Yes @ni3verma

Reply:
There is a problem.
Some fav options opens up the same fragment but with different arguments.
For eg: rotate page, add watermark,etc opens up viewFilesFragment.
How will I set title in such type of cases?

Reply:
@Ni3verma You can use the fragment name i.e. "View Files" for that.

 How about passing existing title as argument when moving to Favorites screen from any section? So that if back button is pressed, the title is sent back.

Reply:
Add password and remove password both opens RemovePagesFragment.
How will I set title in these type of cases?

Reply:
Review my latest commit.
This is the only way I could think to solve it.